test/integration/dataplane_mtls_test.go

```diff
@@ -0,0 +1,182 @@
+		t.Fatalf("Failed to run Docker image: %v", err)
+	}
+	defer cleanupDockerImage(containerName)
+
+	// Step 3: Wait for the Docker container to be ready
+	if err := waitForContainer(containerName); err != nil {
+		t.Fatalf("Container did not start successfully: %v", err)
+	}
+
+	// Step 4: Load TLS config
+	clientTLSConfig, err := setupTLSConfig(TLSConfig{
+		CAFile:   CAFile,
+		CertFile: ClientCertFile,
+		KeyFile:  ClientKeyFile,
+	})
+	require.NoError(t, err)
+
+	// Step 5: Dial grpc to test hand-shake
+	clientCreds := credentials.NewTLS(clientTLSConfig)
+	conn, err := grpc.Dial("localhost:9874", grpc.WithTransportCredentials(clientCreds))
+	require.NoError(t, err)
+	defer conn.Close()
+}
+
+// Helper functions
+func certFile(filename string) string {
+	return filepath.Join("certs", filename)
+	if dir := os.Getenv("TEST_CERTS_PATH"); dir != "" {
+		return filepath.Join(dir, filename)
+	}
+	panic("Env var TEST_CERTS_PATH not found. Please specify path to mTLS test certs")
+}
+
+func setupTLSConfig(cfg TLSConfig) (*tls.Config, error) {
+	var err error
+	tlsConfig := &tls.Config{}
+	if cfg.CertFile != "" && cfg.KeyFile != "" {
+		tlsConfig.Certificates = make([]tls.Certificate, 1)
+		tlsConfig.Certificates[0], err = tls.LoadX509KeyPair(
+			cfg.CertFile,
+			cfg.KeyFile,
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if cfg.CAFile != "" {
+		b, err := ioutil.ReadFile(cfg.CAFile)
+		if err != nil {
+			return nil, err
+		}
+		ca := x509.NewCertPool()
+		ok := ca.AppendCertsFromPEM([]byte(b))
+		if !ok {
+			return nil, fmt.Errorf(
+				"failed to parse root certificate: %q",
+				cfg.CAFile)
+		}
+		if cfg.Server {
+			tlsConfig.ClientCAs = ca
+			tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
+		} else {
+			tlsConfig.RootCAs = ca
+		}
+		tlsConfig.ServerName = cfg.ServerAddress
+	}
+	return tlsConfig, nil
+}
+
+type TLSConfig struct {
+	CertFile      string
+	KeyFile       string
+	CAFile        string
+	ServerAddress string
+	Server        bool
+}
+
+// runCommand is a wrapper around exec.Command to provide more informative
+// errors back to the user
+func runCommand(command string) error {
+	cmd := exec.Command("sh", "-c", command)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Printf("Error executing command: %s\nOutput: %s\n", command, string(output))
+		return err
+	}
+	fmt.Printf("Command succeeded: %s\nOutput: %s\n", command, string(output))
+	return nil
+}
+
+// runDockerImage starts the Docker container with the specified name.
+func runDockerImage(containerName, imageName string) error {
+	cmd := exec.Command(
+		"docker",
+		"run",
+		"--name", containerName,
+		"-d",
+		"-p", "9874:9874",
+		"-v", os.Getenv("PWD")+"/certs:/app/certs",
+		imageName,
+		"mutual-tls",
+		"--server-certificate-path", "/app/certs/server.pem",
+		"--server-private-key-path", "/app/certs/server-key.pem",
+		"--client-certificate-authority-root-path", "/app/certs/root.pem",
+	)
+	return cmd.Run()
+}
```

Comment:
Not using KTF to spin up the dataplane. I believe that's a problem.

Reply:
Right, we use KTF specifically for MetalLB which we use for IPAM for our (bpf hijacked) services right now. So for a complete testing environment we're still a bit bound to KTF (or you can manually reproduce the same environment with your own metallb configmap).